### PR TITLE
Modifying operation handling to support backend changes

### PR DESCRIPTION
**Pull request by ifielker**
Modifying operation handling to support backend changes

### CODE REVIEW

File: firebase_admin/ml.py [hunk 1]
```diff
@@ -841,8 +839,12 @@ def handle_operation(self, operation, wait_for_operation=False, max_time_seconds
                 raise _utils.handle_operation_error(operation.get('error'))
             raise exceptions.UnknownError(message='Internal Error: Malformed Operation.')
 
-        op_name = operation.get('name')
-        _, model_id = _validate_and_parse_operation_name(op_name)
+        op_name = _validate_operation_name(operation.get('name'))
```
Comment (hiranya911) [inline firebase_admin/ml.py]: Nit: Push this further down, as it's not immediately used.

File: firebase_admin/ml.py [hunk 2]
```diff
@@ -841,8 +839,12 @@ def handle_operation(self, operation, wait_for_operation=False, max_time_seconds
...
+        metadata = operation.get('metadata')
+        if (metadata is None or '@type' not in metadata or
+                'ModelOperationMetadata' not in metadata.get('@type')):
+            raise TypeError('Unknown type of operation metadata.')
```
Comment (hiranya911) [inline firebase_admin/ml.py]:
```suggestion
        metadata = operation.get('metadata', {})
        metadata_type = metadata.get('@type', '')
        if not metadata_type.endswith('.ModelOperationMetadata'):
            raise TypeError('Unknown type of operation metadata.')
```

File: firebase_admin/ml.py (hunk 1 above)
Comment (ifielker) [reply firebase_admin/ml.py]: It's an early out. If the operation name is not valid, you won't be able to get_operation below so you might as well not bother with unpacking metadata or setting start and stop times etc. Also, pushing it down would make the testing a bit harder as we'd have to include valid metadata with the op_name validation tests.

File: firebase_admin/ml.py (hunk 2 above)
Comment (ifielker) [reply firebase_admin/ml.py]: Nice! Done.

**Pull request merged**